### PR TITLE
Create lambda to create test data files in Athena output

### PR DESCRIPTION
**Pull request by gdsandrewmccafferty**
This PR introduces a new stack called `audit-test-tools`, which will be used to hold any lambdas etc we specifically need to go into the audit account.

The lambda we're creating here is triggered by a message being posted to the `write-test-data-to-athena-bucket-queue` in the query results account, which we'll be using as part of integration tests for the query-results lambdas.

It will write a "fake" output to the Athena output bucket, and then send a message to the query completed queue, so that code in the results delivery repo triggers. and we can then test the workflow there.

### CODE REVIEW

File: audit-test-tools/src/lambdas/writeTestDataToAthenaBucket/handler.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { SQSEvent } from 'aws-lambda'
```
Comment (rikumolugds) [inline audit-test-tools/src/lambdas/writeTestDataToAthenaBucket/handler.ts:1]: I think there's a second bit this lambda has to do, which is to trigger the query-completed-queue in the audit account. Except if there's something else that already does that.

File: .github/workflows/package-audit-test-tools.yaml [hunk 2]
```diff
@@ -0,0 +1,105 @@
+name: Package dev tools infrastructure
```
Comment (andylangridge) [inline .github/workflows/package-audit-test-tools.yaml:1]: Needs a new name!

File: .github/workflows/package-audit-test-tools.yaml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        with:
+          name: audit-test-tools-sam-template
+          path: |
+            audit-test-tools/dist/
+            audit-test-tools/template.yaml
+
+  package:
+    strategy:
+      matrix:
+        target: [AUDIT_DEV]
```
Comment (andylangridge) [inline .github/workflows/package-audit-test-tools.yaml:67]: You might also want to add AUDIT_BUILD as a target?

```suggestion
        target: [AUDIT_DEV, AUDIT_BUILD]
```

File: audit-test-tools/src/lambdas/writeTestDataToAthenaBucket/handler.ts (hunk 1 above)
Comment (gdsandrewmccafferty) [reply audit-test-tools/src/lambdas/writeTestDataToAthenaBucket/handler.ts:1]: Actually you're right! Totally forgot about. Will sort that out now

File: .github/workflows/package-audit-test-tools.yaml (hunk 3 above)
Comment (gdsandrewmccafferty) [reply .github/workflows/package-audit-test-tools.yaml:67]: As discussed, we've found that this `target` property doesn't actually do anything (by looking at dev-tools and the lack of effect it has there!)

File: .github/workflows/package-audit-test-tools.yaml (hunk 2 above)
Comment (gdsandrewmccafferty) [reply .github/workflows/package-audit-test-tools.yaml]: Done

File: audit-test-tools/template.yaml [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+      Variables:
+        NODE_OPTIONS: --enable-source-maps
+    PermissionsBoundary:
+      !If [UsePermissionsBoundary, !Ref PermissionsBoundary, !Ref AWS::NoValue]
+    ReservedConcurrentExecutions: 10
+    Runtime: nodejs16.x
+    Timeout: 30
+
+Resources:
+  WriteTestDataToAthenaBucketFunction:
```
Comment (andylangridge) [inline audit-test-tools/template.yaml:41]: Do we need an SSM param to output the name so other stacks can use?

Comment (gdsandrewmccafferty) [reply audit-test-tools/template.yaml:41]: No, because we're interacting via a queue

**Review by andylangridge — APPROVED** [review]

File: audit-test-tools/template.yaml [hunk 5]
```diff
@@ -0,0 +1,126 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: Audit accoun test tool infrastructure
```
Comment (rikumolugds) [inline audit-test-tools/template.yaml:2]: Small typo `account`

Comment (gdsandrewmccafferty) [reply audit-test-tools/template.yaml]: Yep, fixed that now

**Review by andylangridge — APPROVED** [review]

File: audit-test-tools/template.yaml [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        Variables:
+          ATHENA_OUTPUT_BUCKET_NAME: '{{resolve:ssm:AthenaQueryOutputBucketName}}'
+          QUERY_COMPLETED_QUEUE_URL: '{{resolve:ssm:QueryCompletedQueueUrl}}'
+      FunctionName: !Sub ${AWS::StackName}-write-test-data-to-athena-bucket
+      KmsKeyArn: '{{resolve:ssm:LambdaKmsKeyArn}}'
+      Events:
+        QRIntegrationTestsTrigggerEvent:
+          Type: SQS
+          Properties:
+            Queue: !Sub 'arn:aws:sqs:${AWS::Region}:{{resolve:ssm:QueryResultsAccountNumber}}:txma-query-results-${Environment}-write-test-data-to-athena-bucket-queue'
```
Comment (rikumolugds) [inline audit-test-tools/template.yaml:59]: Is the queue name correct? I see `txma-qr-infra-build-write-test-data-to-athena-bucket-queue` in the query results build account for example.

Comment (gdsandrewmccafferty) [reply audit-test-tools/template.yaml]: Ah no, you're right, it wasn't! I've corrected it now.

**Review by gds-topickering — APPROVED** [review]

**Review by rikumolugds — APPROVED** [review]

**Pull request merged**